Integrations/freshdesk.yml

```diff
@@ -0,0 +1,345 @@
+    }
+    var auth = 'Basic ' + Base64.encode(params.credentials.identifier + ':' + params.credentials.password);
+
+    var doReq = function(method, path, parameters, body, data) {
+        var result = http(
+            serverUrl + path + encodeToURLQuery(parameters),
+            {
+                Headers: {
+                    'Content-Type': ['application/json'],
+                    'Authorization': [auth]
+                },
+                Method: method,
+                Body: body ? body : '',
+                Data:JSON.stringify(data),
+            }
+        );
+        if (result.StatusCode !== 200 && result.StatusCode !== 201) {
+            throw 'Failed to perform request ' + path + ', request status code: ' + result.StatusCode + ', request body: ' + result.Body;
+        }
+        return result.Body;
+    };
+
+    // All paths that will be used throughout this integration
+    var commandToPath = {
+        'fd-list-tickets': '/helpdesk/tickets/filter/new_and_my_open?format=json',
+        'fd-create-ticket': '/helpdesk/tickets.json',
+        'fd-update-priority': '/helpdesk/tickets/'+parseInt(args.id)+'.json',
+        'fd-add-note': '/helpdesk/tickets/'+parseInt(args.id)+'/conversations/note.json',
+        'fd-update-status': '/helpdesk/tickets/'+parseInt(args.id)+'.json',
+        'fd-assign-ticket': '/helpdesk/tickets/'+parseInt(args.id)+'/assign.json?responder_id='+parseInt(args.userID),
+        'fd-list-agents': '/agents/filter/avtive?format=json',
+        'fd-list-groups': '/groups.json',
+        'fd-get-notes': '/helpdesk/tickets/'+parseInt(args.id)+'.json'
+
+    };
+
+    // Getting the actual display name for users by ID
+    var get_user_by_id = function(id){
+        id = parseInt(id);
+        var res = doReq('GET', '/contacts/'+id+'.json', args);
+        var jsonRes = JSON.parse(res);
+        return jsonRes.user.name;
+
+    };
+
+    // Truncating the output so we don't break markdown
+    var md_trunc = function(str) {
+        str = String(str).replace(/(\r\n|\n|\r)/gm," ");
+        var length = 80;
+        var myString = str;
+        var myTrun = myString.substring(0,length);
+        return myTrun;
+    };
+
+
+
+    switch (command) {
+    // Gets a list of all open tickets and also those that are assigned to the account used
+        case 'fd-list-tickets':
+            var res = doReq('GET', commandToPath[command], args);
+            var jsonRes = JSON.parse(res);
+            results = jsonRes;
+            var md = "#### Freshdesk Open Tickets\n";
+            md += "Ticket ID | Subject | Priority | Status | Assigned To \n-|-|-|-|-\n";
+            for (var i=0;i<results.length;i++) {
+                md += results[i].display_id + "|" + results[i].subject + "|" + results[i].priority_name + "|"+ results[i].status_name + "|" + results[i].responder_name + "\n";
+        }
+        return {Type: entryTypes.note, Contents: results, ContentsFormat: formats.json, HumanReadable: md};
```

Comment:
The retuned object should be split into separate lines.
`return {Type: entryTypes.note, Contents: results, ContentsFormat: formats.json, HumanReadable: md};`
`return {
      Type: entryTypes.note,
      Contents: results,
      ContentsFormat: formats.json,
      HumanReadable: md}
      ;`

Reply:
I'll go over the points as soon as I'm back from a customer visit. 

-Markus